sdk/servicebus/azure-servicebus/tests/test_queues.py

```diff
@@ -2374,4 +2376,90 @@ def test_queue_send_dict_messages_scheduled_error_badly_formatted_dicts(self, se
                     with pytest.raises(TypeError):
                         sender.schedule_messages(message_dict, scheduled_enqueue_time)
                     with pytest.raises(TypeError):
-                        sender.schedule_messages(list_message_dicts, scheduled_enqueue_time)
\ No newline at end of file
+                        sender.schedule_messages(list_message_dicts, scheduled_enqueue_time)
+
+    @pytest.mark.liveTest
+    @pytest.mark.live_test_only
+    @CachedResourceGroupPreparer(name_prefix='servicebustest')
+    @CachedServiceBusNamespacePreparer(name_prefix='servicebustest')
+    @ServiceBusQueuePreparer(name_prefix='servicebustest', dead_lettering_on_message_expiration=True)
+    def test_queue_send_amqp_annotated_message(self, servicebus_namespace_connection_string, servicebus_queue, **kwargs):
+
+        with ServiceBusClient.from_connection_string(
+                servicebus_namespace_connection_string, logging_enable=False) as sb_client:
+            sequence_body = [b'message', 123.456, True]
+            footer = {'footer_key': 'footer_value'}
+            prop = {"subject": "sequence"}
+            seq_app_prop = {"body_type": "sequence"}
+
+            sequence_message = AMQPAnnotatedMessage(
+                sequence_body=sequence_body,
+                footer=footer,
+                properties=prop,
+                application_properties=seq_app_prop
+            )
+
+            value_body = {b"key": [-123, b'data', False]}
+            header = {"priority": 10}
+            anno = {"ann_key": "ann_value"}
+            value_app_prop = {"body_type": "value"}
+
+            value_message = AMQPAnnotatedMessage(
+                value_body=value_body,
+                header=header,
+                annotations=anno,
+                application_properties=value_app_prop
+            )
+
+            data_body = [b'aa', b'bb', b'cc']
+            data_app_prop = {"body_type": "data"}
+            del_anno = {"delann_key": "delann_value"}
+            data_message = AMQPAnnotatedMessage(
+                data_body=data_body,
+                delivery_annotations=del_anno,
+                application_properties=data_app_prop
+            )
+
+            with pytest.raises(ValueError):
+                AMQPAnnotatedMessage(data_body=data_body, value_body=value_body)
+            with pytest.raises(ValueError):
+                AMQPAnnotatedMessage()
+
+            recv_data_msg = recv_sequence_msg = recv_value_msg = 0
+            with sb_client.get_queue_receiver(servicebus_queue.name, max_wait_time=10) as receiver:
+                with sb_client.get_queue_sender(servicebus_queue.name) as sender:
+                    batch = sender.create_message_batch()
+                    batch.add_message(data_message)
+                    batch.add_message(value_message)
+                    batch.add_message(sequence_message)
```

Comment:
since the `create_from_messages_if_dict` now also accepts `messages: List[Union[Mapping, ServiceBusMessage, AMQPAnnotatedMessage]]`, maybe you could add a `dict`/`ServiceBusMessage` to check that a mixed list sends fine here and in async?
(may not really be needed, but just putting it out there)

Reply:
that's a good point, I'll cover it